### ISSUE TITLE
Palworld (1623730)

### ISSUE THREAD

**Issue by ruineka** [ValveSoftware/Proton#7448]
# Compatibility Report
- Palworld
- 1623730

## System Information
- Steam Deck
- Video driver version: Mesa 23.1.3
- Kernel version: 6.1.52-valve14-1-neptune-61
- Proton version: Experimental Bleeding Edge

## I confirm:
- [x ] that I haven't found an existing compatibility report for this game.
- [x ] that I have checked whether there are updates for my system available.

## Symptoms 
The OSK doesn't grab focus of the text field unless you tap in it first. If you manually invoke the keyboard it'll look like it's in focus and receive inputs, but it doesn't. Instead it'll receive gamepad inputs as you navigate through the OSK.

## Reproduction
Go to multiplayer via friend code and attempt to type in a friend code or name your character at the character creation screen.

**Comment by StripedMonkey** [comment]
I'm going to try and debug it here in the next few days and report back, but the game runs perfectly on my desktop (3090, 7950x), but suffers from frequent freezes and appears to hang my Steam Deck when playing for any extended period of time. Looking at protondb, this seems to be a common theme in reports for the deck, though it's hard to say who played long enough to notice the issues.

From personal experience, the most consistent mechanism to induce a freeze is entering and exiting dungeons or loading new regions on the map. Depending on the situation it can be anywhere from 5 minutes to an hour before the deck freezes. Staying roughly in one location/"at base" seems to work fine, but will still eventually cause a freeze.

**Comment by ruineka** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/7448#issuecomment-1962663932

The crashes on the steam deck are a result of the unified memory being maxed out. If you monitor your memory usage you will see the performance drop after 16GB is exceeded and the swapfile is used and once it hits 18GB it will lock the system up. Bosses, dungeons, and frequent fast travels continue to consume more memory and it’s never freed.

**Comment by StripedMonkey** [comment]
Whilst that theory has merits, the fact is that I played for ~9h straight(? It may have been two sessions) on desktop, speed running dungeons across the entire map without issue. If going in and out of a singular dungeon can cap that 18GB limit in 5 minutes, I nor anyone else could play with even 32GB of memory.

It's possible you're right, but I'm not entirely confident that's all that's going on. Unified GPU/CPU memory or not.

**Comment by ruineka** [comment]
> Whilst that theory has merits, the fact is that I played for ~9h straight(? It may have been two sessions) on desktop, speed running dungeons across the entire map without issue. If going in and out of a singular dungeon can cap that 18GB limit in 5 minutes, I nor anyone else could play with even 32GB of memory.
> 
> It's possible you're right, but I'm not entirely confident that's all that's going on. Unified GPU/CPU memory or not.

I have played 200hrs on my Steam Deck exclusively so I know exactly how to crash it and prevent it before it happens. I haven’t played on a desktop, but what I would do is monitor your ram usage on your desktop and see if it goes up over time like you can observe on the Steam Deck. I suspect your desktop would likely drop in performance if the vram is maxed instead of a system lockup like we are seeing with the Steam Deck. 

I think windows will automatically continue to increase the swapfile size over time as well which the Steam Deck will not do.

Let me know the findings on the desktop because it has me curious.

**Comment by StripedMonkey** [comment]
I did not manage to do much in the way of debugging on the SD side of things, but I did spend ~3h this weekend playing Palworld on my desktop. During this session, I fought the penking, chillet, bushi, katress, whatever that lightning horse is, went into 4-5 separate dungeons (with a few dumb deaths that required re-entering the dungeons, this game really needs better armor integrity). Additionally, I found ~30-40 lifmonk statues as a ballpark guesstimate for how much of the map I've covered. The game, by the end of the play session, consumed 14GB of System Memory.

Initially, the game takes 6GB of memory starting at base. 
Entering the Penking dungeon tips it to 6.3GB of memory
Entering the Katress dungeon didn't change the memory usage significantly
Entering the Bushi dungeon upped it to 7GB
Entering the dungeon by the Syndicate tower upped it to 9GB
Exiting the dungeon by the syndicate tower dropped it to 8GB
After this point, I mostly stopped paying attention to the numbers.

These numbers are not representative of 1:1 Steam deck settings, as I'm playing with maxed settings, and was doing lots of other things at the same time which could have influenced overall system usage.

tl;dr, while these numbers could to indicate a memory leak, it does not seem dramatic enough for the "enter a dungeon and immediately freeze" that I've experienced on a fresh restart of the entire steam deck. It also seems plausible that it's only an issue with not freeing assets after being loaded from disk, instead of a true "leak".
Notably, VRAM usage does not change significantly here at all. It peaks upon loading a new area, but quickly goes down so it's almost certainly not a vram issue itself.

**Comment by KrisCris** [comment]
The latest Palworld client running under Proton can no longer connect to any dedicated multiplayer server, just timed out.

**Comment by Maxence1502** [comment]
I confirm it's no longer working since the latest Palworld update

**Comment by simifor** [comment]
@KrisCris @Maxence1502 I am seeing the same with proton stable, but proton 9.0 beta lets me connect. Have you tried with that version?

**Comment by KrisCris** [comment]
> @KrisCris @Maxence1502 I am seeing the same with proton stable, but proton 9.0 beta lets me connect. Have you tried with that version?

I've tried Proton Experiment, Proton 9.0 (Beta), and Proton GE 9-2, but non of them worked for me.

**Comment by simifor** [comment]
@KrisCris upon further testing, proton 8 can also connect to servers, but there are some servers that will time out every time in every configuration I tested, including windows. So I think it's likely that, at least the ones I encountered on my testing, are a server side issue and not proton side.

**Comment by jbdevic** [comment]
> @KrisCris upon further testing, proton 8 can also connect to servers, but there are some servers that will time out every time in every configuration I tested, including windows. So I think it's likely that, at least the ones I encountered on my testing, are a server side issue and not proton side.

I had already tried it on Proton 8 (not working) with a new server, I'm surprised it worked for you

**Comment by matze19999** [comment]
I have two steam decks here, the first one connects to my dedicated server, the second one won't connect. I haven't figured out yet why it is like that

**Comment by simifor** [comment]
Could you add "PROTON_LOG=1 %command%" to the launch parameter of both
steam decks, try to connect to the server, and then upload the resulting
file? It'll create a file in your home folder called steam-1623730.log

**Comment by matze19999** [comment]
> Could you add "PROTON_LOG=1 %command%" to the launch parameter of both steam decks, try to connect to the server, and then upload the resulting file? It'll create a file in your home folder called steam-1623730.log


No problem, here it is:

[steam-1623730_connection_successful.log](https://github.com/ValveSoftware/Proton/files/14999361/steam-1623730_connection_successful.log)
[steam-1623730_no_connection.log](https://github.com/ValveSoftware/Proton/files/14999362/steam-1623730_no_connection.log)

**Comment by alicx1** [comment]
I can confirm that palworld servers dont work when using proton. this is a project that used to work but now no more: https://github.com/Dekita/palhub-server (it may help you to debug)

**Comment by ferreyr9** [comment]
wine without proton works

**Comment by simifor** [comment]
@matze19999 Sorry to ask for new logs, but could you run the game with these options instead `PROTON_LOG=+winhttp,+wininet,+winsock,+secur32,+crypt,+bcrypt,+dnsapi,+steamclient,+cryptnet %command%`. Make sure to not cut the log, if it gets too big, just compress it, even zip should greatly reduce its file size.

**Comment by alicx1** [comment]
> wine without proton works

@BG-Ferreira do you have any link to repo of someone who already did this please?

**Comment by ferreyr9** [comment]
@alicx1 I used this one before i made an pyterodacly egg that works
[docker-palworld-dedicated-server-wine](https://github.com/ripps818/docker-palworld-dedicated-server-wine)

**Comment by matze19999** [comment]
> @matze19999 Sorry to ask for new logs, but could you run the game with these options instead `PROTON_LOG=+winhttp,+wininet,+winsock,+secur32,+crypt,+bcrypt,+dnsapi,+steamclient,+cryptnet %command%`. Make sure to not cut the log, if it gets too big, just compress it, even zip should greatly reduce its file size.

Here are the logs:

[steam-1623730_working.log](https://github.com/ValveSoftware/Proton/files/15203991/steam-1623730_working.log)
[steam-1623730-not_working.log](https://github.com/ValveSoftware/Proton/files/15204014/steam-1623730-not_working.log)

**Comment by simifor** [comment]
@matze19999 I think in this scenario it would help for me to try and connect directly to yor server to try and see what can be found. If you're ok with it, provide details so that I can find it.

**Comment by matze19999** [comment]
> @matze19999 I think in this scenario it would help for me to try and connect directly to yor server to try and see what can be found. If you're ok with it, provide details so that I can find it.

This is a server in a private lan with no access from outsite. But this happens to all servers, also for public server.

**Comment by kisak-valve** [comment]
Palworld (1623730)

Issue transferred from https://github.com/ValveSoftware/Proton/issues/8159.
@Aurontime777 posted on 2024-10-12T15:10:29:

# Compatibility Report
- Name of the game with compatibility issues: Palworld
- Steam AppID of the game: 1623730

## System Information
- GPU: NVIDIA GeForce RTX 3070 Laptop GPU
- Video driver version: NVIDIA 560.35.03
- Kernel version: 6.11.3-200.fsync.fc40.x86_64 (64-bit)
- Link to full system information report as [Gist](https://gist.github.com/Aurontime777/4396eb288c88e7d4860e8bf9a1e90711):
- Proton version: Experimental

## I confirm:
- [ ] that I haven't found an existing compatibility report for this game. (#7892)
- [*] that I have checked whether there are updates for my system available.

Attached Log: 
- Without environment variables: [steam-1623730.log](https://github.com/user-attachments/files/17350510/steam-1623730.log)
- With environment variables: [steam-1623730.log](https://github.com/user-attachments/files/17350456/steam-1623730.log)

## Symptoms

Launching the game without environment variables opens the game normally but it didn't utilize dGPU at all, resulting in 15-30 fps even on 1280*720 lowest settings. Adding `VK_ICD_FILENAMES`, `DXVK_FILTER_DEVICE_NAME`, `force_vk_vendor` variables (both in tandem and in isolation) results in heavy stuttering at the main menu, before eventually crashes (at least in my case, others' cases seemed to work). According to #7892, it might have something to do with GPU enumeration of DirectX 11/12.

## Reproduction

1. Use laptop with multiple GPUs
2. Add environment variables (if you do, go to 4b.)
3. Launch the game (go to 4a.)
4. Encounter the issue symptoms mentioned above.

**Comment by Aurontime777** [comment]
> Link to full system information report as [Gist](https://gist.github.com/Aurontime777/4396eb288c88e7d4860e8bf9a1e90711):

Edit: My system file as Git is here: https://gist.github.com/Aurontime777/4396eb288c88e7d4860e8bf9a1e90711

I didn't see it until the issue is transferred.

**Comment by TheOnlyJoey** [comment]
I am having the same issues as https://github.com/ValveSoftware/Proton/issues/7448#issuecomment-2408598202 where on my laptop with Nvidia GeForce RTX 2070 Super with MaxQ, it always seems to default to the Intel GPU onboard, even when forcing with different type of environment variables. resulting in unusable FPS.
Have tested with Proton Experimental and Proton 9, Driver is Nvidia 545.23.06-3.

I have also tried running in Wine as an alternative (with and without DXVK), that has the same result for me.

Edit: No problem with other games in my library such as Cyberpunk, its just palworld.

**Comment by awsms** [comment]
Looks like rumble doesn't work on this game with Proton? It's enabled in the game settings, I'm using an Xbox One controller (Steam Input is disabled), but no rumble whatsoever.

**Comment by simifor** [comment]
@awsms I couldn't get rumble to work on windows either. Xbox series x controller both wired and wireless, with steam input disabled, no action I performed made the controller rumble. Attacking, being attacked, crafting, falling. Do you know of any specific actions that triggers rumble?

**Comment by FlameSoulis** [comment]
~~Not sure if this is the right spot, but I've noticed issues with Palworld not launching with my laptop's setup (Asus FX505, AMD iGPU and Nvidia RTX 2060 dGPU). Even with various variables thrown, I can't even get the game to launch.~~

EDIT: Posting it here in case it helps someone else: Someone in the Discord suggested running Steam in a terminal to see what is going on. Sure enough, there was a hidden (and helpful!) message:
```
Proton: Upgrading prefix from UMU-Proton-9.0-3.2 to GE-Proton9-22 (/home/flame/.local/share/Steam/steamapps/compatdata/0/)
Proton: Prefix has an invalid version?! You may want to back up user files and delete this prefix.
```
This got me thinking about Palworld's prefix, so I just renamed its folder 1623730 to 1623730.old, relaunched and waited for it to run the setup and cloudsave resyncing. Sure enough, this outright fixed the issue and I can now resume playing Palworld on my laptop.

**Comment by bczhc** [comment]
I'm filing a new issue related to Palworld here:

# Compatibility Report
- Name of the game with compatibility issues: Palworld
- Steam AppID of the game: 1623730

## System Information
- GPU: NVIDIA GeForce RTX 3060 Mobile / Max-Q
- Video driver version: nvidia 590.48.01
- Kernel version: 6.19.0-1-cachyos-bore
- Link to full system information report as [Gist](https://gist.github.com/bczhc/a618f1b3afaf42dd9fb72ed7d4923665):
- Proton version: proton-experimental

[steam-1623730.log.gz](https://github.com/user-attachments/files/25715574/steam-1623730.log.gz)

## Symptoms 

Under Wayland (Hyprland compositor) desktop with a >1.0 scale, on two UIs (Map via 'M' key and the radial menu via '4' key), the cursor speed does not follow system's, just still behaving as if the desktop scaling was 1.0.

For instance, in the game, cursor speed will reset to a slow one upon entering the map zone. Demonstration video:

https://github.com/user-attachments/assets/6e14eac6-8c71-482b-9d9c-8afa7a66fdbf

## Reproduction

1. If Hyprland is used set its monitor scale to a value greater than one. Do something like:

   `hyprctl keyword monitor 'DP-2, 2560x1440@165.00, 0x0, 3.2'`, 

2. Launch the game using proton-experimental.
3. Create/select and enter a save.
4. Press 'M' to open the map.
5. Move around the mouse and observe.

Chart below is the plot of the cursor's x coordinate on my end (Hyprland, 1.3333 scaling). I used `uinput` to signal constant mouse virtual events, letting it move over the map UI (screen leftmost to rightmost). Note the slope of the straight lines at both ends; that's when the cursor isn't in the map area.

<img width="2146" height="834" alt="Image" src="https://github.com/user-attachments/assets/a1b31954-ecb0-483a-b4c6-eb55f6d8c3bc" />

There's another issue btw. Notice the cursor in-game which becomes larger. For this game, desktop scaling should not affect in-game cursor size.

## Additional info

On Windows, cursor speed within the map area is always consistent with what it is outside of this area. That is, if you set Windows Custom Scaling to a higher one, your mouse pointer moves faster, but within the game's map UI, its diamond-shaped cursor also moves faster to act the same as what it is outside. Plus, in-game cursor size won't change regardless of Custom Scaling.

**Comment by matze19999** [comment]
> @matze19999 I think in this scenario it would help for me to try and connect directly to yor server to try and see what can be found. If you're ok with it, provide details so that I can find it.

My server is now available from the Internet. Are you still interested in troubleshooting this?
